core/bug_api.php

```diff
@@ -1545,6 +1550,93 @@ function bug_get_newest_bugnote_timestamp( $p_bug_id ) {
 }
 
 /**
+ * For a list of bug ids, return array of bugnote stats:
+ * the timestamp for the most recent time at which a bugnote
+ * associated with the bug was modified and the total bugnote
+ * count in one db query.
+ * @param array $p_bugs_id Array of Integer representing bug identifiers.
+ * @param integer|null $p_user_id User for checking access levels. null defaults to current user
+ * @return array objects consisting of bugnote stats
+ * @access public
+ * @uses database_api.php
+ */
+function bug_get_bugnote_stats_array( array $p_bugs_id, $p_user_id = null ) {
+	$t_id_array = array();
+	foreach( $p_bugs_id as $t_id ) {
+		$t_id_array[$t_id] = (int)$t_id;
+	}
+	if( empty( $t_id_array ) ) {
+		return array();
+	}
+
+	if ( null === $p_user_id ) {
+		$t_user_id = auth_get_current_user_id();
+	}
+	else {
+		$t_user_id = $p_user_id;
+	}
+
+	db_param_push();
+	$t_params = array();
+	$t_dbparam = array();
```

Comment:
I would suggest a different name than $t_db_param to avoid confusion with $t_param. Maybe $t_in_clause_elements or something similar.